bug(material/list): input coercion for disableRipple on list items does not work

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

Disabling the ripple effect on a `mat-list-item` is currently not possible when using just `disableRipple` on the `mat-list-item` element. The compiler throws the following error: `error TS2322: Type 'string' is not assignable to type 'boolean'.`

It works when using the verbose `[disableRipple]="true"` syntax.

### Reproduction

Steps to reproduce:
1. Add the following snippet to a template
```html
<mat-action-list>
    <button mat-list-item disableRipple>Item</button>
</mat-action-list>
```
2. See the compiler throwing this error: `error TS2322: Type 'string' is not assignable to type 'boolean'.`


### Expected Behavior

It is possible to disable the ripple effect on a `mat-list-item` by just putting `disableRipple` inside the element tag.

### Actual Behavior

The compiler throws an error.

### Environment

Angular CLI: 16.0.2 and 15.2.8
Node: 18.15.0
Package Manager: npm 9.5.0
OS: win32 x64

--- a/src/material/list/list-base.ts
+++ b/src/material/list/list-base.ts
@@ -134,7 +134,7 @@ export abstract class MatListItemBase implements AfterViewInit, OnDestroy, Rippl
       !!this._listBase?.disableRipple
     );
   }
-  set disableRipple(value: boolean) {
+  set disableRipple(value: BooleanInput) {
     this._disableRipple = coerceBooleanProperty(value);
   }
   private _disableRipple: boolean = false;

